Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -283,7 +283,7 @@ open class RouteController: NSObject {
         if (newLegIndex != progress.legIndex) {
             progress.legIndex = newLegIndex
         }
-        if (newStepIndex != progress.currentLegProgress.stepIndex) {
+        if (newStepIndex != progress.currentLegProgress.stepIndex) && (newStepIndex < progress.currentLeg.steps.endIndex - 1) {
```

Comment:
I think it’s premature to apply this defensive fix. We should look into why the navigator thought the new index would be valid even though the route model didn’t have as many steps. This method may turn out to be merely a cascading failure, which we would want to address with an assertion rather than a no-op.

@bamx23, if it would be possible to track down the route request or navigator history for #2980, that would be helpful for this investigation.

Reply:
That’s what @MaximAlien said above in https://github.com/mapbox/mapbox-navigation-ios/pull/2980#issuecomment-841191115; I didn’t see that earlier. I’m just concerned that we’re papering over this crash even further so that the next time, we’ll see a crash or weird behavior when the leg and intersection indices get updated but the step index does not.

Reply:
Yes. I'm worried that if the `stepIndex` didn't get updated but other indexes are updated, there would be some further problems.